google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -0,0 +1,613 @@
+        #   filter results with and return only the specified fields. One or
+        #   more field paths can be specified.
+        #
+        # @return [Query] New query with `select` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.select(:population)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def select *fields
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          fields = fields.flatten.map(&:to_s).reject(&:nil?).reject(&:empty?)
+          new_query.select ||= StructuredQuery::Projection.new
+          fields.each do |field|
+            field = StructuredQuery::FieldReference.new field_path: field.to_s
+            new_query.select.fields << field
+          end
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # Selects documents from all collections, immediate children and nested,
+        # of where the query was created from.
+        #
+        # @return [Query] New query with `all_descendants` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.all_descendants
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def all_descendants
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          if new_query.from.empty?
+            fail "missing collection_id to specify descendants."
+          end
+
+          new_query.from.last.all_descendants = true
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # Selects only documents from collections that are immediate children of
+        # where the query was created from.
+        #
+        # @return [Query] New query with `direct_descendants` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.direct_descendants
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def direct_descendants
```

Comment:
This should be the default.

Reply:
It is the default, but if you previously called `all_descendants` this will change it back. Changed the method to private so it won't appear in the documentation.